### PR TITLE
Added largest_pow_of_two_le_num

### LINKED ISSUE #9347: Add largest power of 2 less than or equal to a given number
### Feature description

This can be done by left-shifting the given number and stopping when you reach the first set bit. And then you can find the largest power of two by finding the difference between the length of the given number when converting it to binary form and the no of times you had to shift to find the first set bit.

### PR DESCRIPTION
**Pull request by namansharma18899**
### Describe your change:
Fixes #9347

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

File: bit_manipulation/largest_pow_of_two_less_than_or_equal_to_a_number.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+For example:
+number: 15
+res:    1   0b1
+        2   0b10
+        4   0b100
+        8   0b1000
+        16  0b10000 (Exit)
+"""
+
+
```
Comment (rahbdd) [inline bit_manipulation/largest_pow_of_two_less_than_or_equal_to_a_number.py:21]: i dont think we need so long one instead we can use 
def find(a):
    if a<=0:
        return 0;
    for i in range(a+1):
        if (2**i <= a):
            c=i;
        else :
            return c;
b=int(input())
print(find(b))

**Review by rahbdd — CHANGES_REQUESTED** [review]
def find(a):
    if a<=0:
        return 0;
    for i in range(a+1):
        if (2**i <= a):
            c=i;
        else :
            return c;
b=int(input())
print(find(b))
 this would be a simple and effective code i guess

File: bit_manipulation/largest_pow_of_two_less_than_or_equal_to_a_number.py (hunk 1 above)
Comment (namansharma18899) [reply bit_manipulation/largest_pow_of_two_less_than_or_equal_to_a_number.py:21]: Hey @rahbdd thanks for the suggestion but it's actually not about the size of the code but how verbose yet briefly it explains the algo.  The actual code I have written is in fact only 5 lines.

I would suggest you to look at all the other algo's in the repository. The goal of the **TheAlgorithms** as described by their authors is _to document model beautiful, helpful and interesting algorithms using code_

File: DIRECTORY.md [hunk 2]
```diff
@@ -51,6 +51,7 @@
   * [Index Of Rightmost Set Bit](bit_manipulation/index_of_rightmost_set_bit.py)
   * [Is Even](bit_manipulation/is_even.py)
   * [Is Power Of Two](bit_manipulation/is_power_of_two.py)
+  * [Is Power Of Two](bit_manipulation/largest_pow_of_two_less_than_or_equal_to_a_number.py)
```
Comment (ChrisO345) [inline DIRECTORY.md:54]: Please __DO NOT__ edit `DIRECTORY.md`. We have a workflow to do this for us.

Comment (namansharma18899) [reply DIRECTORY.md:54]: Understood, I'll revert this change.

File: bit_manipulation/largest_pow_of_two_less_than_or_equal_to_a_number.py (hunk 1 above)
Comment (rahboddeda) [reply bit_manipulation/largest_pow_of_two_less_than_or_equal_to_a_number.py]: Thank you , I will try to modify it

**Review by ChrisO345 — APPROVED** [review]

**Comment by ChrisO345** [comment]
#9374 already implemented.

**Pull request merged**